File: Android/app/src/androidTest/java/jp/co/riso/android/text/IpAddressFilterTest.java

package jp.co.riso.android.text;

import jp.co.riso.smartdeviceapp.SmartDeviceApp;
import junit.framework.TestCase;
import android.text.InputFilter;
import android.widget.EditText;

public class IpAddressFilterTest extends TestCase {
    private static final String INVALID_INPUT_TEXT = "!";
    private static final String VALID_INPUT_TEXT = "a";

    public IpAddressFilterTest(String name) {
        super(name);
    }

    protected void setUp() throws Exception {
        super.setUp();
    }

    protected void tearDown() throws Exception {
        super.tearDown();
    }

    // ================================================================================
    // Tests - constructors
    // ================================================================================

    public void testConstructor() {
        try {
            assertNotNull(new IpAddressFilter());
        } catch (NullPointerException e) {
            fail();
        }
    }

    // ================================================================================
    // Tests - filter
    // ================================================================================

    public void testFilter_Invalid() {
        EditText editText = new EditText(SmartDeviceApp.Companion.getAppContext());
        editText.setFilters(new InputFilter[] {
                new InputFilter.LengthFilter(8),
                new IpAddressFilter()
        });
        editText.setText(INVALID_INPUT_TEXT);
        assertTrue(editText.getText().toString().isEmpty());

    }

    public void testFilter_Valid() {
        EditText editText = new EditText(SmartDeviceApp.Companion.getAppContext());
        editText.setFilters(new InputFilter[] {
                new InputFilter.LengthFilter(8),
                new IpAddressFilter()
        });
        editText.setText(VALID_INPUT_TEXT);
        assertTrue(VALID_INPUT_TEXT.contentEquals(editText.getText()));
    }
}

File: Android/app/src/androidTest/kotlin/jp/co/riso/android/text/IpAddressFilterTest.kt
package jp.co.riso.android.text

import android.text.InputFilter.LengthFilter
import android.widget.EditText
import jp.co.riso.smartdeviceapp.SmartDeviceApp
import junit.framework.TestCase.*
import org.junit.Test

class IpAddressFilterTest {

    // ================================================================================
    // Tests - constructors
    // ================================================================================
    @Test
    fun testConstructor() {
        try {
            assertNotNull(IpAddressFilter())
        } catch (e: NullPointerException) {
            fail()
        }
    }

    // ================================================================================
    // Tests - filter
    // ================================================================================
    @Test
    fun testFilter_Invalid() {
        val editText = EditText(SmartDeviceApp.appContext!!)
        editText.filters = arrayOf(
            LengthFilter(8),
            IpAddressFilter()
        )
        editText.setText(INVALID_INPUT_TEXT)
        assertTrue(editText.text.toString().isEmpty())
    }

    @Test
    fun testFilter_Valid() {
        val editText = EditText(SmartDeviceApp.appContext!!)
        editText.filters = arrayOf(
            LengthFilter(8),
            IpAddressFilter()
        )
        editText.setText(VALID_INPUT_TEXT)
        assertTrue(VALID_INPUT_TEXT.contentEquals(editText.text))
    }

    companion object {
        private const val INVALID_INPUT_TEXT = "!"
        private const val VALID_INPUT_TEXT = "a"
    }
}